dspy/streaming/streaming_listener.py

```diff
@@ -249,10 +249,10 @@ def find_predictor_for_stream_listeners(program: "Module", stream_listeners: lis
                     "predictor to use for streaming. Please specify the predictor to listen to."
                 )
 
-            if field_info.annotation not in [str, Citations]:
+            if not _is_streamable(field_info.annotation):
                 raise ValueError(
-                    f"Stream listener can only be applied to string or Citations output field, but your field {field_name} is of "
-                    f"type {field_info.annotation}."
+                    f"Stream listener can only be applied to string or streamable dspy type output field, "
```

Comment:
Users may get confused by what "streamable dspy type output field" means, maybe something like `subclass of `dspy.Type` that has `is_streamable() == True`?